api_server_new.py

```
@@ -0,0 +1,1354 @@
+                        current_step_index = 0
+                    
+                    # Calculate progress percentage (0-100)
+                    total_steps = len(pipeline_steps)
+                    step_progress = int((current_step_index / total_steps) * 100) if total_steps > 0 else 0
+                    
+                    # Log step progress
+                    logger_task.info(f"Step progress: {step_name} ({current_step_index+1}/{total_steps}): {step_progress}%")
+                    
+                    update_ingest_progress(
+                        "processing",
+                        message=f"Processing file {i+1} of {len(video_files)} - {step_name}",
```

Ruff (0.11.9)

Function definition does not bind loop variable `i`

(B023)